daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -672,23 +680,55 @@ runStart sandboxPort (StartNavigator shouldStartNavigator) (OpenBrowser shouldOp
                   then withNavigator
                   else (\_ _ _ f -> f sandboxPh)
 
-runDeploy :: String -> SandboxPort -> IO ()
-runDeploy host sandboxPort = do
+data HostAndPortFlags = HostAndPortFlags { hostM :: Maybe String, portM :: Maybe Int }
+
+withHostAndPortDefaults :: HostAndPortFlags -> IO HostAndPort
```

Comment:
To me a function called `withX` suggests that it wants a continuation as one of its arguments, most likely the last one. Renaming this to `getX` might be a good idea.

Reply:
ok, will do